app/CMakeLists.txt

```diff
@@ -52,119 +39,94 @@ option(SANITIZE_THREAD    "Sanitize Thread"        OFF)
 option(SANITIZE_UNDEFINED "Sanitize Undefined"     OFF)
 
 # Options for libigl modules
-option(LIBIGL_USE_STATIC_LIBRARY          "Use libigl as static library" OFF)
-option(LIBIGL_WITH_COMISO            "Use CoMiso"                   OFF)
-option(LIBIGL_WITH_EMBREE            "Use Embree"                   OFF)
-option(LIBIGL_WITH_OPENGL            "Use OpenGL"                   OFF)
-option(LIBIGL_WITH_OPENGL_GLFW       "Use GLFW"                     OFF)
-option(LIBIGL_WITH_OPENGL_GLFW_IMGUI "Use ImGui"                    OFF)
-option(LIBIGL_WITH_PNG               "Use PNG"                      OFF)
-option(LIBIGL_WITH_TETGEN            "Use Tetgen"                   OFF)
-option(LIBIGL_WITH_TRIANGLE          "Use Triangle"                 OFF)
-option(LIBIGL_WITH_PREDICATES        "Use exact predicates"         ON)
-option(LIBIGL_WITH_XML               "Use XML"                      OFF)
+option(LIBIGL_USE_STATIC_LIBRARY        "Use libigl as static library" OFF)
+option(LIBIGL_WITH_COMISO            	"Use CoMiso"                   OFF)
+option(LIBIGL_WITH_EMBREE            	"Use Embree"                   OFF)
+option(LIBIGL_WITH_OPENGL            	"Use OpenGL"                   OFF)
+option(LIBIGL_WITH_OPENGL_GLFW       	"Use GLFW"                     OFF)
+option(LIBIGL_WITH_OPENGL_GLFW_IMGUI 	"Use ImGui"                    OFF)
+option(LIBIGL_WITH_PNG               	"Use PNG"                      OFF)
+option(LIBIGL_WITH_TETGEN            	"Use Tetgen"                   OFF)
+option(LIBIGL_WITH_TRIANGLE          	"Use Triangle"                 OFF)
+option(LIBIGL_WITH_PREDICATES        	"Use exact predicates"         ON)
+option(LIBIGL_WITH_XML               	"Use XML"                      OFF)
 
 # Sanitizers
-if(FLOAT_TETWILD_WITH_SANITIZERS)
-	list(APPEND CMAKE_MODULE_PATH ${FLOAT_TETWILD_EXTERNAL}/sanitizers-cmake/cmake)
+if(TWTK_APPS_WITH_SANITIZERS)
+	list(APPEND CMAKE_MODULE_PATH ${TWTK_APPS_EXTERNAL}/sanitizers-cmake/cmake)
 endif()
 
-# Setup dependencies
-# include(FloatTetwildDependencies)
 
 ################################################################################
-# FloatTetwild library
+# FloatTetwild
 ################################################################################
 
-#find_package(GMPfTetWild)
-#IF(NOT ${GMP_FOUND})
-#	MESSAGE(FATAL_ERROR "Cannot find GMP")
-#ENDIF()
-
-# find_package(MPFR)
-# IF(NOT ${MPFR_FOUND})
-# 	MESSAGE(FATAL_ERROR "Cannot find MPFR")
-# ENDIF()
-
-# add_library() can only be called without any source since CMake 3.11 ...
-#set(SOURCES
-#		TetWild.cpp TetWild.h
-#				common.h
-#				Parameters.h
-#				Envelope.h
-#		)
-#
-#prepend_current_path(SOURCES)
-#wildmeshing_toolkit_copy_headers(${SOURCES})
-#wildmeshing_toolkit_set_source_group(${SOURCES})
-#target_sources(${PROJECT_NAME} PRIVATE ${SOURCES})
-
-add_library(${PROJECT_NAME} #SOURCES)
+add_library(TetWild
```

Comment:
Why do you compile it as a library if it has a `main.cpp`? Just use `add_executable()` directly.

Reply:
for the test, i need to link the test target against tetwild

Reply:
An app shouldn't have unit test. If it does, it should be compiled as a proper lib, and not have any `main.cpp`.

Reply:
tetwild is an app, it should be tested, no?

Reply:
uh, sorry you mean that the main shouldn't be there?
If so, it was a mistake sorry